src/lightcurvelynx/utils/extrapolate.py

```diff
@@ -92,34 +96,51 @@ class ZeroPadding(FluxExtrapolationModel):
     def __init__(self):
         super().__init__()
 
-    def _extrapolate(self, last_value, last_flux, query_values):
+    def _extrapolate(self, last_values, last_fluxes, query_values, extrap_axis="time"):
         """Evaluate the extrapolation given the last valid points(s) and a list of new
         query points.
 
         Note
         ----
         This function does not care which axis is being extrapolated. The returned values are
-        always len(query_values) x len(last_flux) and may need to be transposed by the calling
+        always len(query_values) x len(last_fluxes) and may need to be transposed by the calling
         function.
 
         Parameters
         ----------
-        last_value : float
-            The last valid value along the extrapolation axis a which the flux was predicted
+        last_values : float
+            The last valid value along the extrapolation axis at which the flux was predicted
             (e.g., wavelength in AA or time in days).
-        last_flux : numpy.ndarray
+        last_fluxes : numpy.ndarray
             A length N array of the flux values at the last valid time or wavelength (in nJy).
         query_values : numpy.ndarray
             A length M array of values along the extrapolation axis (times in days or wavelengths
             in AA) at which to extrapolate.
+        extrap_axis : str
+            Extrapolation axis. 'time' or 'wavelength'.
 
         Returns
         -------
         flux : numpy.ndarray
             A N x M matrix of extrapolated values. Where M is the number of query points and
             N is the number of flux values at the last valid point.
         """
-        N_len = len(last_flux)
+
+        if extrap_axis not in ["time", "wavelength"]:
+            raise ValueError("Invalid extrap_axis value: only 'time' or 'wavelength' is valid")
+        if not np.isscalar(last_values):
+            if np.any(last_values < query_values):
```

Comment:
To confirm my understanding, this statement is determining whether it is extrapolating before or after. Is that right?

Would it be easier to just add a "before/after" boolean to the extrapolate call?